test/sys/test_ptrace.rs

```diff
@@ -46,6 +47,108 @@ fn test_ptrace_setsiginfo() {
     }
 }
 
+#[test]
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+fn test_ptrace_peekpoke() {
+    use nix::sys::ptrace;
+    use nix::sys::signal::{raise, Signal};
+    use nix::sys::wait::{waitpid, WaitStatus};
+    use nix::unistd::fork;
+    use nix::unistd::ForkResult::*;
+
+    let _m = ::FORK_MTX.lock().expect("Mutex got poisoned by another test");
+
+    // FIXME: qemu-user doesn't implement ptrace on all architectures
+    // and retunrs ENOSYS in this case.
+    // We (ab)use this behavior to detect the affected platforms
+    // and skip the test then.
+    // On valid platforms the ptrace call should return Errno::EPERM, this
+    // is already tested by `test_ptrace`.
+    let err = ptrace::attach(getpid()).unwrap_err();
+    if err == Error::Sys(Errno::ENOSYS) {
+        return;
+    }
+
+    match fork() {
+        Ok(Child) => {
+            ptrace::traceme().unwrap();
+            // As recommended by ptrace(2), raise SIGTRAP to pause the child
+            // until the parent is ready to continue
+            raise(Signal::SIGTRAP).unwrap();
+            unsafe {
+                let size = 10000;
+                let ptr = libc::calloc(size, 1);
+                libc::getcwd(ptr as *mut i8, size);
+                libc::free(ptr);
+                libc::getpriority(0, 42);
+                libc::_exit(0);
+            }
+        }
+        Ok(Parent { child }) => {
+            assert_eq!(
+                waitpid(child, None),
+                Ok(WaitStatus::Stopped(child, Signal::SIGTRAP))
+            );
+
+            let mut syscall_no = None;
+            let mut getpriority_checked = false;
+            let mut getcwd_checked = false;
+
+            ptrace::setoptions(child, ptrace::PTRACE_O_TRACESYSGOOD).unwrap();
+
+            loop {
+                ptrace::syscall(child).unwrap();
+                match waitpid(child, None).unwrap() {
+                    WaitStatus::PtraceSyscall(child) => {
+                        match syscall_no {
+                            None => {
+                                let no = ptrace::peekuser(child, syscall_arg!(0)).unwrap();
```

Comment:
Is it possible to run the tests as a 32-bit process on a 64-bit architecture and so these tests will fail because they use `syscall_arg!`? If so we should either fix the tests to work regardless or disable the tests.

Reply:
Can you address my question here?